src/filtering.rs

```diff
@@ -118,9 +161,114 @@ impl SnapshotFilter {
             }
         }
 
+        if matches!(&self.filter_after, Some(after) if snapshot.time <= after.0)
+            || matches!(&self.filter_before, Some(before) if snapshot.time >= before.0)
+            || matches!((&self.filter_size,&snapshot.summary), (Some(size),Some(summary)) if !size.matches(summary.total_bytes_processed))
+            || matches!((&self.filter_size_added,&snapshot.summary), (Some(size),Some(summary)) if !size.matches(summary.data_added))
+        {
+            return false;
+        }
+
         snapshot.paths.matches(&self.filter_paths)
+            && (self.filter_paths_exact.is_empty()
+                || self.filter_paths_exact.contains(&snapshot.paths))
             && snapshot.tags.matches(&self.filter_tags)
+            && (self.filter_tags_exact.is_empty()
+                || self.filter_tags_exact.contains(&snapshot.tags))
             && (self.filter_hosts.is_empty() || self.filter_hosts.contains(&snapshot.hostname))
             && (self.filter_labels.is_empty() || self.filter_labels.contains(&snapshot.label))
```

Comment:
Same here, a comment with the implied logic would be nice, so we can review against that. Especially because `self.filter_paths_exact.is_empty()` is not negated, which confused me.

Reply:
comments are added.